spec/services/provider_interface/reference_status_report_service_spec.rb

```diff
@@ -0,0 +1,116 @@
+require 'rails_helper'
+
+RSpec.describe ProviderInterface::ReferenceStatusReportService do
+  let(:reference_const) { 'Reference' }
+  let(:references) { [] }
+
+  subject(:reference_service) { described_class.new(references) }
+
+  describe '#headers' do
+    it 'returns the correct headers' do
+      expect(reference_service.headers).to eq([
+        'Name',
+        'Status',
+        'Referee Name',
+        'Application Choice ID',
+        'Link',
+      ])
+    end
+  end
+
+  describe '#reference_link' do
+    it 'returns the correct reference link' do
+      application_choice_id = 123
+      reference = instance_double(reference_const, application_choice_id: application_choice_id)
```

Comment:
we have a [factory](https://github.com/DFE-Digital/apply-for-teacher-training/blob/2612d981958f5e52d1b7f8f25f94e73995676a67/spec/factories/reference.rb) for references – we should use these instead in our tests

Reply:
I realise now it wasn't wholly a reference, it was reference joined on application_form and application_choice so application_choice_id is not on application_reference so I don't think the factory will be appropriate and i may need to stick to making my own.